processor/schemaprocessor/internal/translation/modifier.go

```diff
@@ -0,0 +1,161 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translation // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/translation"
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/alias"
+)
+
+// Modifier abstracts a verion change that allows
+// for the changes to be updated or revert.
+type Modifier interface {
```

Comment:
This interface is probably fine for now, but it likely need to change because we have recently introduced more complicated changes. The Schema File Format 1.1 brought the "split" transformation which changes both the metric name and the attribute name at the same time: https://github.com/open-telemetry/opentelemetry-go/blob/1eae91b3b099af62100cd630cd1cab89fe841615/schema/v1.1/ast/metrics.go#L40